### ISSUE TITLE
Cache can be modified with nested objects

### ISSUE THREAD

**Issue by ste2425** [jpodwys/superagent-cache#54]
**How to reproduce**

- Cache a response with an array.
- Modify modify the response body after it has been cached.
- Make a second network request within the cache expiry.

**Expected behavior**
The cache returns the original cached object before any modifications to the response body took place.

**Actual behavior**
Any changes that were made propagate back to the original object, meaning any subsequent calls get the last modified version of the cache, rather than the original object that was cached.
 
**Example**

```javascript
            superAgent.get('some/url')
                .type('json')
                .end((err, res) => {
                    res.body.length; // 6
                    
                    res.body.splice(0, 1);

                    /*
                        When this is next retrieved from cache the body length with be 1, not 6
                    */
                });
```

I don't believe this behavior is document either.

**Comment by jpodwys** [comment]
This is indeed a problem that exists in [`cache-service-cache-module`](https://github.com/jpodwys/cache-service-cache-module) (which superagent-cache uses) and it has been reported [in this issue](https://github.com/jpodwys/cache-service-cache-module/issues/6).

The issue reporter and I discussed potential complications in solving the problem in the original github issue. I would appreciate if you could add your thoughts there concerning our discussion.

**Comment by jpodwys** [comment]
Fixed in `3.0.0`.
